filecoin-proofs/src/api/update.rs

```diff
@@ -649,3 +714,269 @@ pub fn verify_empty_sector_update_proof<Tree: 'static + MerkleTreeTrait<Hasher =
 
     Ok(valid)
 }
+
+/// Given the specified arguments, this method returns the inputs that were used to
+/// generate the sector update proof.  This can be useful for proof aggregation, as verification
+/// requires these inputs.
+///
+/// This method allows them to be retrieved when needed, rather than storing them for
+/// some amount of time.
+///
+/// # Arguments
+///
+/// * `porep_config` - this sector's porep config that contains the number of bytes in the sector.
+/// * `comm_r_old` - a commitment to a sector's previous replica.
+/// * `comm_r_new` - a commitment to a sector's current replica.
+/// * `comm_d_new` - a commitment to a sector's current data.
+pub fn get_sector_update_inputs<Tree: 'static + MerkleTreeTrait<Hasher = TreeRHasher>>(
+    porep_config: &PoRepConfig,
+    comm_r_old: Commitment,
+    comm_r_new: Commitment,
+    comm_d_new: Commitment,
+) -> Result<Vec<Vec<Fr>>> {
+    trace!("get_sector_update_inputs:start");
+
+    ensure!(
+        comm_r_old != [0; 32],
+        "Invalid all zero commitment (comm_r_old)"
+    );
+    ensure!(
+        comm_r_new != [0; 32],
+        "Invalid all zero commitment (comm_r_new)"
+    );
+    ensure!(
+        comm_d_new != [0; 32],
+        "Invalid all zero commitment (comm_d_new)"
+    );
```

Comment:
@Kubuxu

Reply:
I don't think they are invalid, but I used the [0;32] value in one place as a notion of "default" value in other software. It should not affect this.

Reply:
Looks like these checks are also in for sealing, which we're not going to change in case it had been historically used on mainnet, so having them here also is a consistency check.

Reply:
Should we then add a comment that it's theoretically doing the wrong thing, but we have it there for historical reasons?

Reply:
There's an argument in 'crypto' that the specification is defined by the implementation.  I'd argue it's not doing the wrong thing so I don't think it needs a comment like that.